odbc_fdw.c

```diff
@@ -844,29 +1070,16 @@ static void odbcGetForeignRelSize(PlannerInfo *root, RelOptInfo *baserel, Oid fo
 static void odbcEstimateCosts(PlannerInfo *root, RelOptInfo *baserel, Cost *startup_cost, Cost *total_cost, Oid foreigntableid)
 {
 	unsigned int table_size   = 0;
-	char *svr_dsn             = NULL;
-	char *svr_driver          = NULL;
-	char *svr_host            = NULL;
-	char *svr_port            = NULL;
-	char *svr_database        = NULL;
-	char *svr_schema          = NULL;
-	char *svr_table           = NULL;
-	char *sql_query           = NULL;
-	char *sql_count           = NULL;
-	char *username            = NULL;
-	char *password            = NULL;
-	List *col_mapping_list;
+	odbcFdwOptions options;
```

Comment:
the usage of the structure is pretty inconistent. Here you're "reserving" memory in the stack and not initializing it.